web/packages/api/src/assets_v2.ts

```diff
@@ -612,6 +612,78 @@ export async function fromContext(context: Context): Promise<RegistryOptions> {
     return result
 }
 
+async function checkSnowbridgeV2Support(
+    parachain: ParachainBase,
+    parachainId: number,
+    info: ChainProperties,
+    hasDryRunApi: boolean,
+): Promise<boolean> {
+    console.log(`Checking Snowbridge V2 support for ${info.specName} (parachain ${parachainId})`)
+    if (!hasDryRunApi) {
+        console.error(`  No dry run API available`)
+        return false
+    }
+
+    try {
+        const testAddress = info.accountType === "AccountId32"
+            ? "0x0000000000000000000000000000000000000000000000000000000000000000"
+            : "0x0000000000000000000000000000000000000000"
+
+        const origin = { system: { signed: testAddress } }
+
+        await parachain.provider.call.dryRunApi.dryRunCall(origin, "0x00", 5)
+        console.log(`  XCM V5 supported`)
+
+        if (!isFunction(parachain.provider.call.xcmPaymentApi?.queryXcmWeight)) {
```

Comment:
We should add this as a separate flag `hasXcmPaymentApi` and use it just like you use `hasDryRunApi` check above. Adding this to the registry as a first class property will help debugging.